libraries/EmbeddedChessPieces/src/ECPChessLogic/ECPChessField.h

```diff
@@ -0,0 +1,46 @@
+/**
+ * @file ECPChessField.h
+ * @author Ines Rohrbach, Nico Schramm
+ * @brief Helper classes for chess field
+ * @version 0.1
+ * @date 2024-11-12
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#ifndef ECPChessField_h
+#define ECPChessField_h
+
+#include <Arduino.h>
+
+/**
+ * @brief Enumeration for column of chess board
+ * 
+ * Convertible to int reaching from 0 to 7, i.e. A == 0, B == 1, etc.
+ * 
+ */
+enum ECPBoardColumn {
+    A, B, C, D, E, F, G, H
+};
+
```

Comment:
Not the most elegant solution, but i suggest to add a similar enum to ensure that  the row does not leave the boundary.
At the moment a piece can be placed at { A, 17 }.

```suggestion

/**
 * @brief Enumeration for row of chess board
 * 
 * Convertible to int reaching from 0 to 7, i.e. One == 0, Two == 1, etc.
 * 
 */
enum ECPBoardRow {
    One, Two, Three, Four, Five, Six, Seven, Eight
}

```

Reply:
I've overridden the constructor of `ECPChessField` to validate that the passed `row` is in fact valid. Otherwise a runtime exception will be thrown. That way we can keep the code as readable as possible.

Please see 091303a0a870974bd45393f8878753fb19ae4c1e